### PR TITLE
added a max_depth param to the mem-trackers view

### LINKED ISSUE #7758: [docdb] Add a depth argument to the mem-trackers view
Currently the MemTracker framwork allows us to track memory used across the database. It also allows for arbitrary depth of objects, as we flow the tracking deeper and deeper into various components.

We should add a simple http arg to the `/mem-trackers` UIs, to limit the rendering that we do, to strictly go down to the desired depth.

cc @kmuthukk

### PR DESCRIPTION
**Pull request by yashrsharma44**
Added a max_depth param.
Fixes #7758 
Signed-off-by: Yash Sharma <yashrsharma44@gmail.com>

### CODE REVIEW

**Comment by bmatican** [comment]
Nice quick work @yashrsharma44 !!

Two notes
- did you use some type of auto-format? I see a lot of unrelated format changes
- do you mind posting two screenshots with the default view vs say `max_depth=1` or 2, to confirm the UI would look as expected?

**Comment by yashrsharma44** [comment]
Thanks for the quick response!

> did you use some type of auto-format? I see a lot of unrelated format changes
Yup, I am using the default VSCode formatter, is there any other formatter that you would want me to switch to?

> do you mind posting two screenshots with the default view vs say max_depth=1 or 2, to confirm the UI would look as expected?

Sure!

File: src/yb/server/default-path-handlers.cc [hunk 1]
```diff
@@ -74,7 +74,8 @@
 #include "yb/util/version_info.h"
 #include "yb/util/version_info.pb.h"
 
-DEFINE_int64(web_log_bytes, 1024 * 1024,
+DEFINE_int64(
+    web_log_bytes, 1024 * 1024,
```
Comment (kmuthukk) [inline src/yb/server/default-path-handlers.cc:78]: why do we have this change?

File: src/yb/server/default-path-handlers.cc [hunk 2]
```diff
@@ -118,12 +119,12 @@ struct Tags {
 // Writes the last FLAGS_web_log_bytes of the INFO logfile to a webpage
 // Note to get best performance, set GLOG_logbuflevel=-1 to prevent log buffering
 static void LogsHandler(const Webserver::WebRequest& req, Webserver::WebResponse* resp) {
-  std::stringstream *output = &resp->output;
+  std::stringstream* output = &resp->output;
```
Comment (kmuthukk) [inline src/yb/server/default-path-handlers.cc:122]: prefer not to have unrelated changes in this commit. this could be a separate commit if needed.

**Review by kmuthukk — COMMENTED** [review]
some cosmetic comments. (thx by the way for working on this).

File: src/yb/server/default-path-handlers.cc (hunk 1 above)
Comment (yashrsharma44) [reply src/yb/server/default-path-handlers.cc:78]: Apologies for the unrelated changes, it was the auto-formatter that made those changes, I will revert the changes back 😅

File: src/yb/server/default-path-handlers.cc (hunk 2 above)
Comment (yashrsharma44) [reply src/yb/server/default-path-handlers.cc:122]: Same as above

File: src/yb/server/default-path-handlers.cc [hunk 3]
```diff
@@ -184,10 +184,20 @@ static void MemTrackersHandler(const Webserver::WebRequest& req, Webserver::WebR
   *output << "<table class='table table-striped'>\n";
   *output << "  <tr><th>Id</th><th>Current Consumption</th>"
       "<th>Peak consumption</th><th>Limit</th></tr>\n";
+  
+  int max_depth = INT_MAX;
+  string depth = FindWithDefault(req.parsed_args, "max_depth", "false");
+  if (depth != "false") {
```
Comment (bmatican) [inline src/yb/server/default-path-handlers.cc:190]: maybe make the default just empty string, so we don't hardcode a "false" string in the if check as well

File: src/yb/server/default-path-handlers.cc [hunk 4]
```diff
@@ -184,10 +184,20 @@ static void MemTrackersHandler(const Webserver::WebRequest& req, Webserver::WebR
...
+    max_depth = std::stoi(depth);
+  }
 
   std::vector<MemTrackerData> trackers;
   CollectMemTrackerData(MemTracker::GetRootTracker(), 0, &trackers);
   for (const auto& data : trackers) {
+    // if the data.depth >= max_depth, skip the info
```
Comment (bmatican) [inline src/yb/server/default-path-handlers.cc:197]: nit: Capitalize If and . at the end of comment

**Review by bmatican — COMMENTED** [review]
Small comments.

If you could upload the 2 screenshots I had asked to the main GH issue as well, I could probably approve and merge this afterwards!

**Comment by yashrsharma44** [comment]
> If you could upload the 2 screenshots I had asked to the main GH issue as well, I could probably approve and merge this afterwards!

Hey, do I need to build the binary to run the UI? Because the build process is taking a lot of time.

File: src/yb/server/default-path-handlers.cc (hunk 3 above)
Comment (yashrsharma44) [reply src/yb/server/default-path-handlers.cc]: Sure

File: src/yb/server/default-path-handlers.cc (hunk 4 above)
Comment (yashrsharma44) [reply src/yb/server/default-path-handlers.cc]: Okay

**Comment by yashrsharma44** [comment]
@bmatican Does this works as you expected?

`max_depth=0`
image

`max_depth=1`
image

`max_depth=2`
image

**Comment by bmatican** [comment]
Niiice, this looks great! Let me run this through our CI, just to be safe. I'll merge it in assuming everything is ok!

**Comment by yashrsharma44** [comment]
> Niiice, this looks great! Let me run this through our CI, just to be safe. I'll merge it in assuming everything is ok!

Sure, I am in no hurry, just wanted to make sure if its working properly :)

**Review by bmatican — APPROVED** [review]
